nautobot/core/middleware.py

```diff
@@ -22,14 +34,38 @@ def header(self):
         return settings.REMOTE_AUTH_HEADER
 
     def process_request(self, request):
-
         # Bypass middleware if remote authentication is not enabled
-        if not settings.REMOTE_AUTH_ENABLED:
+        if not remote_auth_enabled(auth_backends=settings.AUTHENTICATION_BACKENDS):
             return
 
         return super().process_request(request)
 
 
+class ExternalAuthMiddleware(MiddlewareMixin):
```

Comment:
I'm not familiar with `MiddlewareMixin`, but from a quick skim of the Django docs it sounds like it exists mostly for providing backwards-compatibility with much older versions of Django. None of our other middleware classes in this module are using it - is it actually needed here?

Reply:
I have no idea; I haven't used Django in years.  Please provide a functioning alternative.

Reply:
You can just ditch the base class. It only needs to conform to the public  interface https://docs.djangoproject.com/en/3.1/topics/http/middleware/#writing-your-own-middleware